### PR TITLE
Add Sage embed

### PR DESCRIPTION
**Pull request by mouryab**

### CODE REVIEW

**Review by girish-thoughtspot — APPROVED** [review]

File: src/embed/sage.spec.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        document.body.innerHTML = getDocumentBody();
+    });
+
+    test('should render sage', async () => {
+        const sageEmbed = new SageEmbed(getRootEl(), defaultConfig);
+        sageEmbed.render();
+        await executeAfterWait(() => {
+            expectUrlMatchesWithParams(
+                getIFrameSrc(),
+                `http://${thoughtSpotHost}/?embedApp=true&hideEurekaResults=false&isSageEmbed=true&disableWorksheetChange=false&hideWorksheetSelector=true&hideEurekaSuggestions=true&hideAction=%5B"save","pin","editACopy","saveAsView","updateTSL","editTSL","onDeleteAnswer","share"%5D#/`,
```
Comment (ashubham) [inline src/embed/sage.spec.ts:41]: add the path after #

File: src/react/index.tsx [hunk 2]
```diff
@@ -150,6 +151,14 @@ export const SearchBarEmbed = componentFactory<
     SearchBarViewConfig
 >(_SearchBarEmbed);
 
+interface SageEmbedProps extends EmbedProps, SageViewConfig {}
+
+export const SageEmbed = componentFactory<
```
Comment (ashubham) [inline src/react/index.tsx:156]: Lets finalize the name.

Comment (mouryab) [reply src/react/index.tsx:156]: Confirmed with Jasmeet

File: src/embed/sage.spec.ts (hunk 1 above)
Comment (mouryab) [reply src/embed/sage.spec.ts]: Updated in newer version

**Pull request closed without merging**